zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs

```diff
@@ -416,37 +459,55 @@ mod tests {
             witnesses: &[U256],
             region: &mut CachedRegion<'_, '_, F>,
         ) -> Result<(), Error> {
-            let [base_fee, fee_overhead, fee_scalar] = [0, 1, 2].map(|i| witnesses[i].as_u64());
+            let [is_curie, base_fee_before_curie, fee_overhead, fee_scalar, tx_data_gas_cost, tx_l1_fee_before_curie] =
+                [0, 1, 2, 3, 4, 5].map(|i| witnesses[i].as_u64());
+
+            let [base_fee_after_curie, l1_blob_basefee, commit_scalar, blob_scalar, tx_signed_length, tx_l1_fee_after_curie] =
+                [6, 7, 8, 9, 10, 11].map(|i| witnesses[i].as_u64());
 
             let l1_fee = TxL1Fee {
-                chain_id: eth_types::forks::SCROLL_MAINNET_CHAIN_ID,
-                block_number: 1,
-                base_fee,
+                chain_id: eth_types::forks::SCROLL_DEVNET_CHAIN_ID,
+                // block_number 5 is starting number for curie in test devnet.
+                block_number: if is_curie == 1 { 5 + 1 } else { 1 },
+                base_fee: if is_curie == 1 {
+                    base_fee_after_curie
+                } else {
+                    base_fee_before_curie
+                },
                 fee_overhead,
                 fee_scalar,
-                l1_blob_basefee: 1,
-                commit_scalar: 0,
-                blob_scalar: 0,
+                l1_blob_basefee,
+                commit_scalar,
+                blob_scalar,
             };
-            let tx_data_gas_cost = witnesses[3];
             self.gadget.assign(
                 region,
                 0,
                 l1_fee,
                 TxL1Fee::default(),
-                tx_data_gas_cost.as_u64(),
-                0, // TODO: check if need update here
+                tx_data_gas_cost,
+                tx_signed_length,
             )?;
+
             self.tx_data_gas_cost.assign(
                 region,
                 0,
                 Value::known(tx_data_gas_cost.to_scalar().unwrap()),
             )?;
-            self.expected_tx_l1_fee.assign(
-                region,
-                0,
-                Value::known(witnesses[4].to_scalar().unwrap()),
-            )?;
+
+            self.is_curie
+                .assign(region, 0, Value::known(F::from(is_curie)))?;
+            let rlp_signed_len = if is_curie == 1 { tx_signed_length } else { 0 };
+            self.tx_signed_length
+                .assign(region, 0, Value::known(F::from(rlp_signed_len)))?;
```

Comment:
i think it is ok to still assign tx_signed_length even when is_curie == 0?    Not used anyway

Reply:
yes

Reply:
in `begin_tx` , this `tx_signed_length` is  looked up from tx table, so can not be any value eventually in real circuit.